scripts/o3de/tests/test_engine_template.py

```diff
@@ -297,6 +342,70 @@ def instantiate_template_wrapper(self, tmpdir, create_from_template_func, instan
 
             return test_folder
 
+    #test create Remote repo
+    @pytest.mark.parametrize(
+        "force, expect_results",
+         [
+            pytest.param(True, 0),
+            pytest.param(False, 0)
+        ]
+    )
+    def test_create_repo(self, tmpdir, force, expect_results):
+        concrete_contents = TEST_REPO_JSON
+        templated_contents = TEST_TEMPLATE_REPO_JSON 
+        template_json_contents = TEST_TEMPLATE_REMOTEREPO_JSON_CONTENTS
+
+        # Append the project.json to the list of files to copy from the template
+        template_json_dict = json.loads(template_json_contents)
+        template_json_dict.setdefault('copyFiles', []).append(
+        {
+            "file": "repo.json",
+            "isTemplated": True
+        })
+
+        # Convert the python dictionary back into a json string
+        template_json_contents = json.dumps(template_json_dict, indent=4)
+
+        instantiated_name = 'TestRepo'
+        #  Use a SHA-1 Hash of the destination_name for every Random_Uuid for determinism in the test
+        concrete_contents = string.Template(concrete_contents).safe_substitute(
+            {
+                'Random_Uuid': str(uuid.uuid5(uuid.NAMESPACE_DNS, instantiated_name)).upper()
+            })
+
+        engine_root = (pathlib.Path(tmpdir) / 'engine-root').resolve()
+        engine_root.mkdir(parents=True, exist_ok=True)
+
+        template_default_folder = engine_root / 'Templates'
+        template_default_folder.mkdir(parents=True, exist_ok=True)
+
+        template_json = template_default_folder / 'template.json'
+        with template_json.open('w') as s:
+            s.write(template_json_contents)
+        
+        file_template_path = template_default_folder / 'Template' / "repo.json"
+        file_template_path.parent.mkdir(parents=True, exist_ok=True)
+        with file_template_path.open('w') as file_template_handle:
+            file_template_handle.write(templated_contents)
+
+        template_dest_path = engine_root / instantiated_name
+       
+        # Skip registration in test
+        with patch('uuid.uuid4', return_value=uuid.uuid5(uuid.NAMESPACE_DNS, instantiated_name)) as uuid4_mock, \
+                patch('o3de.manifest.load_o3de_manifest', return_value={}) as load_o3de_manifest_patch, \
+                patch('o3de.manifest.save_o3de_manifest', return_value=True) as save_o3de_manifest_patch:
+            result = engine_template.create_repo(template_dest_path, repo_name="TestRepo", repo_uri = "http://test", summary="summary", origin='o3de',
```

Comment:
If you take my suggestion above then you'll need to add the `.com` so the uri is a valid URI
```suggestion
            result = engine_template.create_repo(template_dest_path, repo_name="TestRepo", repo_uri = "http://test.com", summary="summary", origin='o3de',
```